digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/MassTransitStopService.scala

```diff
@@ -513,20 +519,31 @@ trait MassTransitStopService extends PointAssetOperations {
   }
 
   private def getPersistedStopWithPropertiesAndPublishEvent(assetId: Long, roadLinkByLinkId: Long => Option[VVHRoadlink],
-                                                            operation: Operation,
-                                                            relevantToTR: Boolean = false) = {
+                                                            operation: Operation = Operation.Noop) = {
     val persistedStop = fetchPointAssets(withId(assetId)).headOption
 
-    if (relevantToTR) {
-      val roadLink = roadLinkByLinkId.apply(persistedStop.get.linkId)
+    persistedStop.foreach { stop =>

```

Comment:
Why does this have foreach? Isn't there only one asset returned with headOption?